hipscat/margin_utils.py

```diff
@@ -0,0 +1,195 @@
+#
+# Healpix utilities
+#
+
+import numpy as np
+import healpy as hp
+
+# see the documentation for get_edge() about this variable
+_edge_vectors = [
+    np.asarray([1, 3]), # NE edge
+    np.asarray([1]),    # E corner
+    np.asarray([0, 1]), # SE edge
+    np.asarray([0]),    # S corner
+    np.asarray([0, 2]), # SW edge
+    np.asarray([2]),    # W corner
+    np.asarray([2, 3]), # NW edge
+    np.asarray([3])     # N corner
+]
+
+# cache used by get_margin()
+_suffixes = {}
+
+def get_edge(dk, pix, edge):
+    # Given a pixel pix of at some order, return all
```

Comment:
I'm wondering if we should have a README in /src/hipscat/pixel_math to explain various algorithms in more detail (reading it in code comments is kinda tricky). or /docs/src/pixel_math.rst.

Side point: I think anything that gets fiddly should probably go in the shared hipscat package.

Reply:
for sure on both